src/expr/src/scalar/mod.rs

```diff
@@ -1399,6 +1399,11 @@ pub enum EvalError {
     IntervalOutOfRange,
     TimestampOutOfRange,
     CharOutOfRange,
+    IndexOutOfRange {
+        provided: i32,
+        // The last valid index position, i.e. `v.len() - 1`
+        valid_end: i32,
```

Comment:
Is `u64` a safer choice here?

Reply:
Ditto the previous comment; also, this must return -1 somewhere for PG compat, so just stuck it on the type to avoid doing computation when printing the error message.